cvat-canvas3d/src/typescript/canvas3dView.ts

```diff
@@ -750,12 +754,27 @@ export class Canvas3dViewImpl implements Canvas3dView, Listener {
         return cuboid;
     }
 
-    private setupObjects(): void {
+    private setupObjects(selected?: boolean): void {
+        const { clientID } = this.model.data.activeElement;
         if (this.views.perspective.scene.children[0]) {
-            this.clearSceneObjects();
+            if (!selected) {
+                this.clearSceneObjects();
+            }
+            this.resetColor();
             this.setHelperVisibility(false);
             for (let i = 0; i < this.model.data.objects.length; i++) {
                 const object = this.model.data.objects[i];
+                if (selected) {
+                    const target = this.views.perspective.scene.getObjectByName(clientID);
```

Comment:
``clientID`` is not changed in the loop, why do we get ``target`` identifier in the loop?

Reply:
If target is const, than line: 

```this.views.perspective.scene.children[0].remove(target);`` tries to remove the same element a lot of times.
Please, correct me if I am wrong